packager/common.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Google Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#ifndef PACKAGER_COMMON_H_
+#define PACKAGER_COMMON_H_
+
+#include <type_traits>
+
+namespace shaka {
+
+/// A mix-in to disable copying and assignment.
+/// Usage: class Foo : private DisallowCopyAndAssign
+class DisallowCopyAndAssign {
+ public:
+  DisallowCopyAndAssign(const DisallowCopyAndAssign&) = delete;
+  DisallowCopyAndAssign& operator=(const DisallowCopyAndAssign&) = delete;
+
+ protected:
+  DisallowCopyAndAssign() = default;
+  ~DisallowCopyAndAssign() = default;
+};
+
+/// A work-alike for Chromium base's arraysize macro, but built on the C++11
+/// standard-library's less-easy-to-understand std::extent template.
+#define arraysize(a) std::extent<decltype(a)>::value
```

Comment:
Oh, nice.  I never found that mentioned in any of the absl docs.

If you don't object, though, I will define our macro here to refer to ABSL_ARRAYSIZE, rather than using ABSL_ARRAYSIZE directly.  There are currently 478 references to arraysize() in the codebase, and I'd rather not replace them all right now.

If you would rather see arraysize() replaced with ABSL_ARRAYSIZE, I will file an issue and do that as one big automated cleanup at the end.  Just let me know what you think.

Reply:
That is fine with me.